tv_apps/smart-home/js/card_picker.js

```diff
@@ -68,10 +124,20 @@
         if (this.mode == 'update') {
           this._showButton(this.selected.length ? 'done' : 'remove');
         }
+      } else if (elem === this.input) {
+        if (this.isKeyboardOpened) {
+          this.closeKeyboard();
+        } else {
+          this.openKeyboard();
+        }
       }
     },
 
     onKeyUp: function(evt) {
+      if (SharedUtils.isBackKey(evt) && this.isKeyboardOpened) {
```

Comment:
Could you merge these two `if` to avoid calling `SharedUtils.isBackKey(evt)` twice?